Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java

```diff
@@ -193,6 +193,44 @@ public void setSelectedPhysicalNode(TreeNode selectedPhysicalNode) {
         return Optional.ofNullable(dataObject instanceof MediaUnit ? (MediaUnit) dataObject : null);
     }
 
+    /**
+     * Select given MediaUnit in physical structure tree.
+     *
+     * @param mediaUnit
+     *          MediaUnit to be selected in physical structure tree
+     */
+    public void selectMediaUnit(MediaUnit mediaUnit) {
+        TreeNode matchingTreeNode = getMatchingTreeNode(getPhysicalTree(), mediaUnit);
+        if (Objects.nonNull(matchingTreeNode)) {
+            updatePhysicalNodeSelection(matchingTreeNode);
+            matchingTreeNode.setSelected(true);
+        }
+    }
+
+    private TreeNode getMatchingTreeNode(TreeNode parent, MediaUnit mediaUnit) {
+        TreeNode matchingTreeNode = null;
+        for (TreeNode treeNode : parent.getChildren()) {
+            if (Objects.nonNull(treeNode.getData()) && treeNode.getData() instanceof StructureTreeNode) {
+                StructureTreeNode structureTreeNode = (StructureTreeNode) treeNode.getData();
+                if (Objects.nonNull(structureTreeNode.getDataObject()) && structureTreeNode.getDataObject() instanceof MediaUnit) {
```

Comment:
`                if (structureTreeNode.getDataObject() instanceof MediaUnit) {`

instanceof returns false for null, so both checks can be done in one statement